routesrv/eskipbytes.go

```diff
@@ -39,7 +44,10 @@ func (e *eskipBytes) formatAndSet(routes []*eskip.Route) (int, bool) {
 
 	e.mu.Lock()
 	defer e.mu.Unlock()
-	e.data = buf.Bytes()
+	if !bytes.Equal(e.data, buf.Bytes()) {
+		e.data = buf.Bytes()
+		e.etag = fmt.Sprintf(`"%s"`, strconv.FormatUint(uint64(crc32.ChecksumIEEE(e.data)), 10))
```

Comment:
I think 
```go
e.etag = fmt.Sprintf(`"%d"`, crc32.ChecksumIEEE(e.data))
```
should do

Reply:
BTW what is the reason to use crc32 over e.g. md5 or some other cryptographic hash?
I think md5 is quite popular for the purpose:
- https://cloud.google.com/storage/docs/hashes-etags
- https://docs.aws.amazon.com/AmazonS3/latest/API/API_Object.html

Reply:
We did some internal benchmarking and found `crc32` to be the fastest.
Here is the benchmarking stats
`BenchmarkDiff-16             357       3171983 ns/op     1434948 B/op      36495 allocs/op // MD5
BenchmarkDiffCrc-16          416       2913816 ns/op     1434699 B/op      36494 allocs/op // IEEE
BenchmarkDiffCrc-16          423       2839012 ns/op     1434719 B/op      36494 allocs/op // Castagnoli
BenchmarkDiffCrc-16          100       10309766 ns/op    2872003 B/op      37901 allocs/op // Koopman
BenchmarkDiffAdler-16        410       2923837 ns/op     1434849 B/op      36494 allocs/op // Adler
BenchmarkDiffFnv-16          392       2902272 ns/op     1434974 B/op      36495 allocs/op // Fnv
BenchmarkDiffMaphash-16      378       2944291 ns/op     1474711 B/op      36509 allocs/op // Maphash `

Hence, Castagnoli or crc32 seems to be the fastest.
But I don't mind switching to `md5`. Let me know your preference

Reply:
Although the performance difference does not look dramatic, lets use crc32 for now.

Reply:
Hmm crc32 is normally used to detect bit flips in network packet size of data. I think the problem can be that it will show the same output for different routes.
I would say let's use sha256 and call it a day. What we really want is a func that does it uniquely as possible, fast is not very important, it's counted in µs anyways, so who cares about this?
It's an optimization that does not change your hot data path.

Reply:
Generating ETag is tricky because there is a tradeoff between generating uniquely(correctness) vs (memory + CPU +  time) performance.

Reply:
Sure, lets use sha256.